packages/dds/tree/docs/storage/compatibility.md

```diff
@@ -0,0 +1,273 @@
+# Compatibility
+
+This document provides concrete guidelines and strategies for organizing code that impacts SharedTree's persisted format.
+It then pivots to the current state of SharedTree's persisted format, and outlines steps to take to align the implementation
+with those guidelines.
+
+## Prerequisites
+
+[This document](../../../SchemaVersioning.md) provides general "best practices" for working with persisted data within the Fluid Framework.
+It's strongly recommended to read through and understand its rationale before continuing with this document,
+as most of the concrete recommendations presented henceforth fall out of those best practices.
+
+## What State is Persisted?
+
+A DDS's persisted format encompasses the format it uses for its summaries as well as its ops (due to [trailing ops](../../../README.md))
+including transitively referenced structured blob data.
+
+Since documents are stored outside of Fluid control (i.e. no type of central data migration is possible),
+DDSes necessarily commit to backwards compatibility of their format for all time.
```

Comment:
I think this is too strong of a statement. Imaging that DDS is used only in scenarios of transient (non-persistent) containers.
I think what you want to say is that general purpose DDSs should follow that policy, as we do not control or know how data is stored and used over time.

Reply:
Maybe worth clarifying what I've written above: I'm reading this document as a guidance to _any_ DDS author. Maybe it's wrong, but it feels like we need to write general-purpose document, and then maybe a clarification document for SharedTree?

Reply:
The document as a whole is specific to SharedTree, but you're right I aimed for this section to be more general before getting into the weeds. I'll update to mention your point.